library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -2619,11 +2625,16 @@ public AudioTrackScore(Format format, Parameters parameters, @Capabilities int f
       this.language = normalizeUndeterminedLanguageToNull(format.language);
       isWithinRendererCapabilities =
           isSupported(formatSupport, /* allowExceedsCapabilities= */ false);
-      preferredLanguageScore =
-          getFormatLanguageScore(
-              format,
-              parameters.preferredAudioLanguage,
-              /* allowUndeterminedFormatLanguage= */ false);
+      int bestLanguageScore = 0;
+      for (int i = 0; i < parameters.preferredAudioLanguage.length; i++) {
+        int score = getFormatLanguageScore(
+            format,
+            parameters.preferredAudioLanguage[i],
+            /* allowUndeterminedFormatLanguage= */ false);
+        score = 1000 * score + parameters.preferredAudioLanguage.length - i; // Priorise the first items in array
```

Comment:
Can you use the same pattern as below for the system locale languages that works without magic score calculation? You can save the best score so far and replace the best match only if you find a better score.